pages/doing_api_first_development.md

```diff
@@ -11,96 +11,337 @@ sitemap:
+
+## Using the openapi-generator-maven-plugin
+
+The OpenAPI Specification (OAS) file is read by the generator from the defined file in the plugin's `inputSpec` element. JHipster's default location for the OAS is `src/main/resources/swagger/api.yml`. The plugin generates interfaces and Data Transfer Objects (DTOs) from the OAS for implementation. The interfaces have default methods which respond with a `501 Not implemented` HTTP status and an empty body.
+
+### Developing an API with the OpenAPI Specification 
+
+This guide will use the classic Pet Store specification from OpenAPI for simplicity as writing an OAS is beyond the scope of this guide. API developers can write an OAS and place it in `src/main/resources/swagger/api.yml` using a tool such as [swagger-editor](http://editor.swagger.io). Many IDEs provide OAS tools for editing as well.
+
+#### Pro Tip
+
+JHipster, when choosing `API first development using OpenAPI-generator`, provides a Docker Compose descriptor for the Swagger Editor at `src/main/docker/swagger-editor.yml`. To use this editor locally, run `docker compose -f src/main/docker/swagger-editor.yml up -d`.
+
+### JHipster's Layered Architecture and Technical Structure Tests by ArchUnit
+
+Prior to implementing the delegates generated by the OpenAPI Generator plugin, take a moment to review and understand JHipster's layered architecture design. This architecture is tested by [TNG Tech's ArchUnit](https://www.archunit.org/motivation) and defined in `src/test/java/com/mycompany/myapp/TechnicalStructureTest.java`. Proper implementation of the delegate classes must be followed so as not to violate this layered architecture.
+
+## Implementation Guide
 
-In API first development, instead of generating the documentation from the code, you need to write the specification first and then generate code from it.
-This has the following advantages:
+### The Pet Store v3
 
-- You can design your API for the consumers and not as a consequence of your implementation.
-- You can use the specification file to mock your new server endpoints before they are released so you can more decouple frontend and backend development.
-- You don't need a live server to use your OpenAPI documentation.
+Hereafter, this guide will be based on a [JHipster-ready version of Expanded Pet Store v3](api/api.yml) from the OpenAPI Initiative ([found here](https://github.com/OAI/OpenAPI-Specification/blob/main/examples/v3.0/petstore-expanded.yaml)). This JHipster-version sets the `server:url` to the JHipster defaults of `http://localhost:8081/api` (where the port defaults to `8081` and the generated controller has a base path of `/api`, matching the default `SecurityConfiguration#filterChain(HttpSecurity,  MvcRequestMatcher.Builder)` implementation).
```

Comment:
Is it 8081?

Reply:
When generating a new `microservice`, the default is 8081:

```shell
? As you are running in a microservice architecture, on which port would like your server to run? It should be unique to
 avoid port conflicts. (8081) 
```